docs(mesh): update ECS docs for new workload identity feature (#4717)
Kong/docs.konghq.com: src/mesh/installation/ecs.md

The control plane itself also runs as an ECS service in the cluster.

### Data plane authentication

As part of joining and synchronizing with the mesh, every sidecar needs to authenticate with
the control plane.

> With Kong Mesh, this is typically accomplished by using a data plane token.

Reviewer: ```suggestion
With {{site.mesh_product_name}}, this is typically accomplished by using a data plane token.
```

Author: I've opened another PR for this as it's unrelated to this change

Author: https://github.com/Kong/docs.konghq.com/pull/4721

Kong Mesh runs in Universal mode on ECS. The example setup repository uses an AWS RDS
database as a PostgreSQL backend. It also uses ECS service discovery to enable ECS
tasks to communicate with the Kong Mesh control plane.

The example Cloudformation includes two Cloudformation stacks for
[creating a cluster](https://github.com/Kong/kong-mesh-ecs/blob/workload-identity/deploy/vpc.yaml) and
[deploying Kong Mesh](https://github.com/Kong/kong-mesh-ecs/blob/workload-identity/deploy/controlplane.yaml)

> #### Workload identity

Reviewer 2: As all this is new should it be guarded with `{% if_version gte:2.0.x %}` ? @cloudjumpercat to help :)

Reviewer: Yep, anything that is specific to 2.0 will need `{% if_version gte:2.0.x %}` and `{% endif_version %}` around it. More info here: https://github.com/Kong/docs.konghq.com/blob/main/docs/single-sourced-versions.md#conditional-rendering

Reviewer: And anything that was removed for 2.0 but exists in previous versions should have `{% if_version lte:1.9.x %}` and `{% endif_version %}` around it instead of removing it completely from the doc

Author: I've just created two entire pages for each version.

  Value: aws-iam
- Name: KUMA_DP_SERVER_AUTH_USE_TOKEN_PATH
  Value: "true"
- Name: KMESH_AWSIAM_AUTHORIZEDACCOUNTIDS
  Value: !Ref AWS::AccountId # this tells the CP which accounts can be used by DPs to authenticate
```

Every sidecar must have the [`--auth-type=aws` flag set as well](https://github.com/Kong/kong-mesh-ecs/blob/workload-identity/deploy/counter-demo/demo-app.yaml#L255).

## Services

When deploying an ECS task to be included in the mesh, the following must be
considered.

### Outbounds

Services are bootstrapped with a `Dataplane` specification.

Transparent proxy is not supported on ECS, so the `Dataplane` resource for a
service must enumerate all other mesh services this service contacts and include them
[in the `Dataplane` specification as `outbounds`][dpp-spec].

See the example repository to learn
[how to handle the `Dataplane` template with Cloudformation](https://github.com/Kong/kong-mesh-ecs/blob/workload-identity/deploy/counter-demo/demo-app.yaml#L31-L46).

### IAM role

The ECS task IAM role must also have some tags set in order to authenticate.
It must always have the `kuma.io/type` tag set to either `"dataplane"`,
`"ingress"`, or `"egress"`.

> If it's a `"dataplane"` type then it must also have the `kuma.io/mesh` tag set.

Reviewer: ```suggestion
If it's a `"dataplane"` type, then it must also have the `kuma.io/mesh` tag set.
```

> Additionally, it can have the `kuma.io/service` tag set to further restrict its identity.

Reviewer: ```suggestion
Additionally, you can set the `kuma.io/service` tag to further restrict its identity.
```
